File: resource-registry/src/main/java/nl/haploid/octowight/registry/repository/ResourceRootDmoRepository.java
package nl.haploid.octowight.registry.repository;

import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface ResourceRootDmoRepository extends MongoRepository<ResourceRootDmo, Long> {

	ResourceRootDmo findByResourceTypeAndAtomIdAndAtomTypeAndAtomOrigin
			(final String resourceType, final Long atomId, final String atomType, final String atomOrigin);

	List<ResourceRootDmo> findByResourceType(final String resourceType);

	ResourceRootDmo findByResourceTypeAndResourceId(final String resourceType, final long resourceId);
}

File: resource-registry/src/main/scala/nl/haploid/octowight/registry/repository/ResourceRootDmoRepository.scala
package nl.haploid.octowight.registry.repository

import java.{lang, util}

import org.springframework.data.mongodb.repository.MongoRepository

trait ResourceRootDmoRepository extends MongoRepository[ResourceRootDmo, lang.Long] {

  def findByResourceType(resourceType: String): util.List[ResourceRootDmo]

  def findByResourceTypeAndRootIdAndRootCategoryAndRootOrigin(resourceType: String, atomId: lang.Long, atomCategory: String, atomOrigin: String): ResourceRootDmo

  def findByResourceTypeAndTombstone(resourceType: String, tombstone: Boolean): util.List[ResourceRootDmo]

  def findByResourceTypeAndResourceId(resourceType: String, resourceId: lang.Long): ResourceRootDmo
}
